app.py

```diff
@@ -115,119 +119,32 @@ def get_frame_range_from_shotgun(self):
 
         return ( data[sg_in_field], data[sg_out_field] )
 
-
     def get_current_frame_range(self, engine):
-
-        if engine == "tk-maya":
-            import pymel.core as pm
-            import maya.cmds as cmds
-            current_in = cmds.playbackOptions(query=True, minTime=True)
-            current_out = cmds.playbackOptions(query=True, maxTime=True)
-
-        elif engine == "tk-nuke" and not self.engine.hiero_enabled:
-            import nuke
-            current_in = int(nuke.root()["first_frame"].value())
-            current_out = int(nuke.root()["last_frame"].value())
-
-        elif engine == "tk-motionbuilder":
-            from pyfbsdk import FBPlayerControl, FBTime
-
-            lPlayer = FBPlayerControl()
-            current_in = lPlayer.LoopStart.GetFrame()
-            current_out = lPlayer.LoopStop.GetFrame()
-
-        elif engine == "tk-softimage":
-            import win32com
-            xsi = win32com.client.Dispatch('XSI.Application')
-
-            current_in = xsi.GetValue("PlayControl.In")
-            current_out = xsi.GetValue("PlayControl.Out")
-
-        elif engine == "tk-houdini":
-            import hou
-            current_in, current_out = hou.playbar.playbackRange()
-
-        elif engine == "tk-3dsmax":
-            from Py3dsMax import mxs
-            current_in = mxs.animationRange.start
-            current_out = mxs.animationRange.end
-        elif engine == "tk-3dsmaxplus":
-            import MaxPlus
-            ticks = MaxPlus.Core.EvalMAXScript("ticksperframe").GetInt()
-            current_in = MaxPlus.Animation.GetAnimRange().Start() / ticks
-            current_out = MaxPlus.Animation.GetAnimRange().End() / ticks
-
-        else:
-            raise tank.TankError("Don't know how to get current frame range for engine %s!" % engine)
-
-        return (current_in, current_out)
+        try:
```

Comment:
For both get and set frame range, rather than returning a `True/False` status, I think it would be great (and more pythonic) to just raise an exception within the hook code and then catch that inside the app - this would allow us to display an error message to the user to explain _why_ the frame range couldn't be set in the case of the set frame range method failing (rather than just displaying `There was a problem updating your scene frame range.`).